app/Domain/Auth/UserAware.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Domain\Auth;
+
+use App\Domain\Users\Models\UserModel;
+use Auth0\Laravel\Auth\Guard;
+use Illuminate\Contracts\Auth\Factory;
+
+trait UserAware
+{
+    public function getUser(): UserModel
+    {
+        /** @var Factory $auth */
+        $auth = auth();
```

Comment:
I agree that it would be better to inject the dependency via the constructor. Optionally if you don't want to repeat the `$auth->guard('auth0')->user()` in every class, you can wrap the `Illuminate\Contracts\Auth\Factory` in some kind of custom `CurrentUserRepository` with a `getCurrentUser()` class that does this for you, so you only need to call the `getUser()` on that class like you do now on `$this` using the trait.

Reply:
I removed `UserAware` and replace with with `AuthController` with injected `Auth` facade.

Reply:
That works and it's okay for me in this particular case as it's not really that important right now, but usually composition is advised over inheritance https://en.wikipedia.org/wiki/Composition_over_inheritance

In this case it's kind of composition because you inject the `Auth` facade, but by adding the base class with `getUser()` you also introduce inheritance which has more limitations (you can only inherit from one class). 

That's why I suggested to wrap the factory (or auth facade, that works as well) in a custom class with a simple `getUser()` that would contain essentially the same logic as the `getUser()` in the base class. Then you just need to inject it and do something like `$this->auth->getUser()`, which is not really that more complicated then `$this->getUser()`. You'd also avoid the `parent::__construct()` call.

Reply:
It already implemented the default Laravel HTTP controller, that's why I created the `AuthController`

I don't mind creating an extra class to get the current user, but the inheritance will be needed.

Reply:
Yeah you're right about the default Laravel HTTP controller and that's okay, that's something that comes with the choice of using a framework like this so I'm not going to argue against that. But extra abstract base controllers is going to run into limitations at some point because you cannot combine them.

Reply:
I did a few extra tweaks:
- I moved the `UserModel` to the `Auth` domain folder
- I created a `CurrentUser` wrapper class and removed the `AuthController` abstract class